core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -82,6 +85,27 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
 
   backfillPrewarms(true)
 
+  // check periodically every 1 minute, delete prewarmed container if unused for sometime
+  context.system.scheduler.schedule(10.seconds, 1.minute) {
+    prewarmConfig.foreach { config =>
+      val kind = config.exec.kind
+      val memory = config.memoryLimit
+      val ttlMinutes = config.ttlMinutes
+      val containers = prewarmedPool.filter { warmInfo =>
+        warmInfo match {
+          case (_, PreWarmedData(_, `kind`, `memory`, _)) => true
+          case _                                          => false
+        }
+      }
+      for ((container, data) <- containers) {
+        if (Duration.between(data.lastUsed, Instant.now).compareTo(Duration.ofMinutes(ttlMinutes)) > 0) {
```

Comment:
If there is no activation for prewarm kind for the given TTL time, it will always cause a cold start next time.
Also, even if we can add more prewarm containers when cold starts happen I think it may not fit well for some workloads. If loads spike for some runtime kind, it will cause a lot of cold starts. Also, it will try to create more prewarm containers at the same time. As we know, docker operation is quite expensive and it will end up with a huge number of container creation along with long wait-time.

Reply:
Yes, agree with you.

I am also thinking about it how to avoid `cold start` after ttl time and new activations come.
One option is:
Use `minCount` and `maxCount`  together instead of `count`
minCount >= 0, maxCount >=minCount  (it is decided by user)

When invoker starts up, creates `maxCount` prewarmed containers.
afer ttl time, can delete some unused containers but need to keep the number of prewarm containers equal with `minCount`

Reply:
@style95 already fixed